routers/record.go

```diff
@@ -0,0 +1,56 @@
+package routers
+
+import (
+	"strings"
+
+	"github.com/astaxie/beego/context"
+	"github.com/casdoor/casdoor/object"
+	"github.com/casdoor/casdoor/util"
+)
+
+func getUser(ctx *context.Context) (username string) {
+	defer func() {
+		if r := recover(); r != nil {
+			username = getUserByClientIdSecret(ctx)
+		}
+	}()
+
+	username = ctx.Input.Session("username").(string)
```

Comment:
Can we use `GetSessionUser` here?

I added some codes to `GetSessionUser`, if user login expired, this function will automatically remove the session data...

Reply:
Regarding GetSessionUser PR has not been merged yet, it cannot be used. At the same time, GetSessionUser is a method of c, but ctx does not have this method.

Reply:
Got that. I will solve the problem.

Plz add Casbin License, then the PR will be OK.